### PR TITLE
Tasks 1-5 done, all errors and warnings removed

### PR DESCRIPTION
**Pull request by VedantKalbag**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -99,6 +93,24 @@ float CentralProcessingUnit::processData()
     return processedData;
 }
 
+struct cpuCounter
```
Comment (matkatmusic) [inline main.cpp]: type names need to begin with a capital letter
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -99,6 +93,24 @@ float CentralProcessingUnit::processData()
...
+{
+    CentralProcessingUnit cpu1;
+    int startCounter(int initVal, int maxVal)
+    {
+        int i = 0;
+        cpu1.counter = initVal;
+        while (i < maxVal)
```
Comment (matkatmusic) [inline main.cpp]: you never increment `i`, so the only thing making this loop exit is the check on line 106.
- [ ] revise.  increment `i` or change your `while( expr )`

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Demonstrate some for() loops.  you've only demonstrated while() loops.
come up with 2 for() loops, and 1 more while() loop.
make the innards of your loops print out something that shows some counting happening.
i.e.
```
for( int i = 0; i < 3; ++i )
    std::cout << "i's value: " << i << std::endl;
```
It doesn't have to be amazing.  but I need you to demonstrate to me that you know how to use these language elements correctly.

File: main.cpp [hunk 3]
```diff
@@ -93,23 +94,40 @@ float CentralProcessingUnit::processData()
...
+            std::cout<<cpu1.counter<<"\n";
             if (cpu1.counter == maxVal)
                 return cpu1.counter;
         }
-        return cpu1.counter;
+    return 0;
     }
     
 };
+struct cpuReset //While #1
```
Comment (matkatmusic) [inline main.cpp]: Type names need to be Capitalized.
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -458,6 +478,19 @@ Library::Library()
     lateFees = 2.5f;
 }
 
+std::string buyNewBooks (int numNewBooks) //While #2
```
Comment (matkatmusic) [inline main.cpp]: That's a nice **free** function you have created here!
It's a shame it's using variables that aren't declared inside it.
It's also interesting that you have a member function declared in `Library`, but not implemented anywhere
🤔 🤔 🤔 🤔 
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 4 above)
Comment (VedantKalbag) [reply main.cpp]: I don't fully understand what you mean here:
1) What is a free function?
2) I am trying to change a member variable of the same struct (that may not be what is asked here, but I would like to understand the right way to do that)
3) Yes, I declared it and defined the function, but replit showed me there was an error so I never implemented it, but was trying to fix the error- which was where I was stuck with point 2) and needed your help here!

Comment (matkatmusic) [reply main.cpp:481]: A free function is a function that is not a member function.  it is not part of a class.
is `buyNewBooks` supposed to be a member of your `Library` class?   if so, you better name it correctly so it is!

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**